### PR TITLE
Xpost Notifications: show `Follow conversation by email` on original post

### LINKED ISSUE #17158: Xpost Notifications: Follow conversation by email not shown on original post
### Expected behavior
When visiting a post from a xpost notification, the `Follow conversation by email` option should be available.

### Actual behavior
It is not.

### Steps to reproduce the behavior
- On two P2s you are following, ensure mobile notifications are enabled for new posts:

<kbd>Screen Shot 2021-09-15 at 11 30 59 AM</kbd>

- On site A, make a post cross posting to site B.
- In the app, go to Notifications. You should have two: 
  - one for the original post on site A
  - one for the cross post on site B
- Select the Notification for the original post.
- Tap the Comments button at the bottom.
- Notice the `Follow(ing) conversation by email` button appears in the header:

<kbd>Screen Shot 2021-09-15 at 11 35 52 AM</kbd>

- Go back and select the Notification for the xpost.
- In the body of the Notification, tap the link to the original post.
- Tap the Comments button at the bottom.
- Notice the `Follow(ing) conversation by email` button does not appear in the header:

<kbd>Screen Shot 2021-09-15 at 11 38 39 AM</kbd>

### PR DESCRIPTION
**Pull request by ScoutHarris**
Fixes #17158
WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/446

When viewing the original post from an xpost Notification, the `Follow conversation by email` button now appears in the Comments header.

When the link is tapped in the Notification body, it now compares the `post.crossPostMeta.postURL` with the tapped URL. If they are the same, the linked post is shown using the IDs from `post.crossPostMeta` (i.e. the original post). This allows the endpoint to be called that returns `subscribe` information for the original post, thus having the necessary information to display the follow button.

Bonus: fixed issues that were causing `Implicit conversion from enumeration type 'enum UIStackViewAlignment' to different enumeration type 'UIStackViewDistribution' (aka 'enum UIStackViewDistribution')` warnings in `ReaderPostHeaderView`.

To test:
- On a P2 you are following, ensure mobile notifications are enabled for new posts:

<kbd>notif_settings</kbd>

- On a different P2, make a post cross posting to the P2 above.
- In the app, go to Notifications.
- Select the Notification for the xpost.
- In the body of the Notification, tap the link to the original post.
- Tap the Comments button at the bottom.
- Verify the `Follow(ing) conversation by email` button appears in the header and shows the correct state.

| Before | After |
|--------|-------|
| Screen Shot 2021-09-15 at 11 38 39 AM | Screen Shot 2021-09-15 at 4 36 49 PM |

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested different Notification types to ensure posts are loaded correctly.

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailCoordinator.swift [hunk 1]
```diff
@@ -494,10 +495,41 @@ class ReaderDetailCoordinator {
...
     ///
     private func presentReaderDetail(_ url: URL) {
-        let readerDetail = ReaderDetailViewController.controllerWithPostURL(url)
+
+        // If the user tapped the link to the original post, use the original post's info to display reader detail.
+        // The API endpoint used by controllerWithPostID returns subscription flags for the post.
+        // The API endpoint used by controllerWithPostURL does return this information.
+        // These flags needed to display the `Follow conversation by email` option.
+        // So if we can call controllerWithPostID, do so. Otherwise, fallback to controllerWithPostURL.
+        // Ref: https://github.com/wordpress-mobile/WordPress-iOS/issues/17158
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailCoordinator.swift]: Should we preface this comment with something making it clear it's regarding cross posts?
We might be missing a word or two in these comments. 
Should `The API endpoint used by controllerWithPostURL does return this information` read `does not`? 
Should `These flags needed to display the `Follow conversation by email` option` say `These flags are`?

File: WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailCoordinator.swift [hunk 2]
```diff
@@ -494,10 +495,41 @@ class ReaderDetailCoordinator {
...
+
+        let readerDetail: ReaderDetailViewController = {
+            if let post = post,
+               selectedUrlIsCrossPost(url) {
+                return ReaderDetailViewController.controllerWithPostID(post.crossPostMeta.postID, siteID: post.crossPostMeta.siteID)
+            }
+
+            return ReaderDetailViewController.controllerWithPostURL(url)
+        }()
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailCoordinator.swift:514]: Nifty!

File: WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailCoordinator.swift [hunk 3]
```diff
@@ -494,10 +495,41 @@ class ReaderDetailCoordinator {
...
+
         viewController?.navigationController?.pushViewController(readerDetail, animated: true)
     }
 
+    private func selectedUrlIsCrossPost(_ url: URL) -> Bool {
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailCoordinator.swift:519]: I'll be "that guy" and ask if we should add a unit test for this?

**Review by aerych — CHANGES_REQUESTED** [review]
Hola @ScoutHarris 👋 
Works great! :) 
Tested as described (thanks for the tip on how to get the notification) and everything worked as expected.  I just had a couple of nitpicky comments. :)

File: WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailCoordinator.swift (hunk 1 above)
Comment (ScoutHarris) [reply WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailCoordinator.swift]: What? You got the gist right? 😉 Thank you, I'll fix it.

File: WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailCoordinator.swift (hunk 3 above)
Comment (ScoutHarris) [reply WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailCoordinator.swift:519]: Oh I suppose we could. 😄 I can do that in a follow up PR if you don't mind.

**Comment by ScoutHarris** [comment]
Hey @aerych .

I updated the comments. I'll do a follow up PR for the unit test, and I'll fix the merge conflict when I merge my WPKit change. 

Back to you please!

File: WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailCoordinator.swift (hunk 3 above)
Comment (aerych) [reply WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailCoordinator.swift:519]: wfm!

**Review by aerych — APPROVED** [review]
:shipit: with great 🚢 ing!

**Comment by ScoutHarris** [comment]
Thanks @aerych !

**Pull request merged**